code/espurna/utils.ino

```diff
@@ -346,6 +346,50 @@ void info() {
 
 }
 
+void info_device(JsonObject& root) {
+    char chipid[7];
+    snprintf_P(chipid, sizeof(chipid), PSTR("%06X"), ESP.getChipId());
+
+    uint8_t * bssid = WiFi.BSSID();
+    char bssid_str[20];
+    snprintf_P(bssid_str, sizeof(bssid_str),
+        PSTR("%02X:%02X:%02X:%02X:%02X:%02X"),
+        bssid[0], bssid[1], bssid[2], bssid[3], bssid[4], bssid[5]
+    );
+
+    root["app_name"] = APP_NAME;
+    root["app_version"] = APP_VERSION;
+    root["app_build"] = buildTime();
+    root["app_revision"] = APP_REVISION;
+    root["manufacturer"] = MANUFACTURER;
+    root["chipid"] = String(chipid);
+    root["mac"] = WiFi.macAddress();
+    root["bssid"] = String(bssid_str);
+    root["channel"] = WiFi.channel();
+    root["device"] = DEVICE;
+    root["hostname"] = getSetting("hostname");
+    root["network"] = getNetwork();
+    root["deviceip"] = getIP();
```

Comment:
If you get  the API answer, you ALREADY know the IP address, don't you?

Reply:
Yeah, but that would mean additional step to get (m)DNS response?

Reply:
I'm not really used with mDNS timing because I use a BIND dns server (with 2 slaves) at home ...yes, i'm a geek ;-)...and in this case query times are totally negligibles. Anyway mDNS queries should not be that long too... whenever someone really need to resolve the IP. 

I'm just trying to save on precious ESP memory.